litebird_sim/bandpasses.py

```diff
@@ -0,0 +1,275 @@
+        mask_below = np.ma.masked_inside(self.freqs, f_below, self.f0).mask
+        x_bel = np.linspace(f_below, self.f0, self.freqs[mask_below].size)
+        self.weights[mask_below] = apod(x_bel, self.f0, f_below)
+
+    # Chebyshev profile bandpass
+    def get_chebyshev_bandpass(self, order=3, ripple_dB=3, normalize=False):
+        """
+        Define a bandpass with chebyshev prototype.
+
+        Args:
+        - order (int): chebyshev filter order
+        - ripple_dB (int): maximum ripple amplitude in decibels
+
+        If order and ripple_dB are not specified a value of 3 is used for both.
+
+        """
+        b, a = signal.cheby1(
+            order,
+            ripple_dB,
+            [2.0 * np.pi * self.f0 * 1e9, 2.0 * np.pi * self.f1 * 1e9],
+            "bandpass",
+            analog=True,
+        )
+        w, h = signal.freqs(b, a, worN=self.freqs * 2 * np.pi * 1e9)
+
+        self.weights = abs(h)
+        if normalize:
+            A = self.get_normalization()
+            self.weights /= A
+            self.isnormalized = True
+
+    def get_normalization(self):
+        """
+        Estimate the integral over the frequency band
+        """
+        return np.trapz(self.weights, self.freqs)
+
+    # Find effective central frequency of a bandpass profile
+    def find_central_frequency(self):
+        """Find the effective central frequency of
+        a bandpass profile as defined in https://arxiv.org/abs/1303.5070
+        """
+        if self.isnormalized:
+            return np.trapz(self.freqs * self.weights, self.freqs)
+        else:
+            return (
+                np.trapz(self.freqs * self.weights, self.freqs)
+                / self.get_normalization()
+            )
+
+    @staticmethod
+    def from_imo(imo: Imo, url: Union[UUID, str]):
+        """Create a `BandPassInfo` object from a definition in the IMO
+        The `url` must either specify a UUID or a full URL to the
+        object.
+
+        """
+        obj = imo.query(url)
+        return BandPassInfo.from_dict(obj.metadata)
+
+    def interpolate_band(self):
+        """
+        This function aims at building the sampler in order to generate random samples
+        statistically equivalent to the model bandpass
+        """
+        # normalize band
+
+        if not self.isnormalized:
+            self.normalize_band()
+        # Interpolate the band
+        b = sp.interpolate.interp1d(x=self.freqs, y=self.weights)
+        # estimate the CDF
+        Pnu = np.array(
+            [
+                sp.integrate.quad(b, a=self.freqs.min(), b=inu)[0]
+                for inu in self.freqs[1:]
+            ]
+        )
+        # interpolate the inverse CDF
+        self.Sampler = sp.interpolate.interp1d(
+            Pnu,
+            self.freqs[:-1] + np.diff(self.freqs),
+            bounds_error=False,
+            fill_value="extrapolate",
+        )
+
+    def bandpass_resampling(self, bstrap_size=1000, nresample=54, model=None):
```

Comment:
The docstring should mention what `model` is supposed to be. (Better also to add a type hint, e.g., `model: Optional[str] = None`.)